### PR TITLE
CLDC-2242 Add assigned to filter

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**
Adds an assigned to filter (like the previous user filter, but with the option to select a specific user as well as all/you)

image

ticket: https://dluhcdigital.atlassian.net/browse/CLDC-2242

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -144,9 +144,9 @@ def assignable_roles
 
   def logs_filters(specific_org: false)
     if (support? && !specific_org) || organisation.has_managing_agents?
-      %w[status years user organisation bulk_upload_id]
+      %w[status years assigned_to user organisation bulk_upload_id]
```
Comment (kosiakkatrina) [inline app/models/user.rb]: do we still need user here?

File: app/helpers/filters_helper.rb [hunk 2]
```diff
@@ -42,6 +43,11 @@ def organisations_filter_options(user)
     [OpenStruct.new(id: "", name: "Select an option")] + organisation_options.map { |org| OpenStruct.new(id: org.id, name: org.name) }
   end
 
+  def assigned_to_filter_options(user)
+    user_options = user.organisation.users
+    [OpenStruct.new(id: "", name: "Select an option")] + user_options.map { |user_option| OpenStruct.new(id: user_option.id, name: user_option.name) }
+  end
+
```
Comment (kosiakkatrina) [inline app/helpers/filters_helper.rb]: Do we want this to show all the users for support user?

File: app/models/user.rb (hunk 1 above)
Comment (natdeanlewissoftwire) [reply app/models/user.rb]: Yep - just to distinguish between the "you" selection or the "specific user" selection with the current user selected (both would have the same value for user, but are distinguishable when we also include the assigned_to value

File: app/helpers/filters_helper.rb (hunk 2 above)
Comment (natdeanlewissoftwire) [reply app/helpers/filters_helper.rb]: Hm good point - I imagine so but I'll check and update the ACs to include that if so :)

Comment (natdeanlewissoftwire) [reply app/helpers/filters_helper.rb]: have made this change, just waiting to hear back re whether we should also include non-support users' managing agents'/stock owners' users

**Review by kosiakkatrina — APPROVED** [review]

File: app/helpers/filters_helper.rb (hunk 2 above)
Comment (natdeanlewissoftwire) [reply app/helpers/filters_helper.rb]: we should, and now we do ^

**Pull request merged**